imports/plugins/included/taxes-taxcloud/client/settings/taxcloud.js

```diff
@@ -20,8 +22,23 @@ Template.taxCloudSettings.helpers({
 AutoForm.hooks({
   "taxcloud-update-form": {
     onSuccess: function () {
-      return Alerts.toast(i18next.t("admin.taxSettings.shopTaxMethodsSaved"),
-        "success");
+      if (!TaxCodes.findOne({ taxCodeProvider: "taxes-taxcloud" })) {
+        Meteor.call("taxcloud/getTaxCodes", (err, res) => {
+          if (err) {
+            throw new Meteor.Error(500, "description");
+          } else if (res && Array.isArray(res)) {
```

Comment:
I'm a bit skeptical that this code is doing what you want.

There are a few problems I want to point out:

1. You're toasting success here, and then looping through the responses and potentially throwing an error. It seems contradictory to alert the user of success if the error processing hasn't happened yet.

2. If you throw an error in this `forEach` loop, the loop will stop executing in the middle. You're not catching the error anywhere, and the error will not `toast` alert the user, it will just stop the app from continuing and log the error in the console.

Reply:
I realised it wasn't necessary to be checking for errors while looping so I have refactored my code to work better.